src/server/middleware/openInEditorMiddleware.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ *
+ * @flow
+ *
+ * --- OVERVIEW ---
+ *
+ * React Native apps show a stack trace when an error occurs. Each
+ * frame can be tapped on by the user. When they do, React Native will
+ * send a POST to /open-stack-frame passing along {file, lineNumber}.
+ *
+ * Haul should then open the user's editor to the file and lineNumber.
+ *
+ * What makes this gloriously fragile is, people use different editors
+ * on different platforms.
+ *
+ * We will rely on the NPM package `open-in-editor` to provide this
+ * functionality. (read: scapegoat).
+ */
+import type { $Request, $Response, Middleware } from 'express';
+import type { ReactNativeStackFrame } from '../../types';
+
+// A configuration object fed into `open-in-editor`.
+type OpenInEditorConfig = {
+  editor?: string | null,
+  cmd?: string | null,
+  pattern?: string | null,
+};
+
+const basename = require('path').basename;
+const openInEditor = require('open-in-editor');
+const editors = Object.keys(require('open-in-editor/lib/editors'));
+const messages = require('../../messages');
+const logger = require('../../logger');
+
+// does `open-in-editor` know how to detect this editor?
+const isKnownEditor = (editor: string): boolean =>
+  editors.indexOf(basename(editor)) >= 0;
+
+/**
+ * Detect the editor type to use. We use just the basename in case the user
+ * has passed a full path. Just the name work here though. Examples:
+ *
+ *   - sublime
+ *   - atom
+ *   - /usr/local/bin/atom
+ *
+ */
+function detectEditorType(): string | null {
+  // check for HAUL_EDITOR first
+  if (process.env.HAUL_EDITOR) {
```

Comment:
yeah, reading env variable for editor seems reasonable.

do we need a separate `HAUL_EDITOR` though? Won't you use the same one you use for React anyway?